pcsx2/PAD/Linux/InputManager.cpp

```diff
@@ -0,0 +1,88 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2021  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "InputManager.h"
+#include "Device.h"
+#include "keyboard.h"
+#include "state_management.h"
+
+#ifdef SDL_BUILD
+#include "SDL/joystick.h"
+#endif
+
+std::unique_ptr<InputDeviceManager> device_manager(new InputDeviceManager);
+
+InputDeviceManager::InputDeviceManager()
+{
+}
+
+InputDeviceManager::~InputDeviceManager()
+{
+	device_manager->devices.clear();
```

Comment:
Any reason this references `device_manager` instead of `this`?  It's crashing on shutdown with clang's libc++ because unique_ptr's destructor copies the pointer into a temporary and wipes the original before calling the destructor.